File: androidApp/src/main/kotlin/com/mohamedfaridelsherbini/moventiq/ui/onboarding/OnboardingUiState.kt
package com.mohamedfaridelsherbini.moventiq.ui.onboarding

data class OnboardingUiState(
    val currentPage: Int = 0,
    val hasCompletedOnboarding: Boolean? = null,
    val isFinished: Boolean = false,
) {
    val shouldShowOnboarding: Boolean
        get() = hasCompletedOnboarding == false && !isFinished

    val currentPageData: OnboardingPage
        get() = OnboardingPage.entries[currentPage.coerceIn(0, OnboardingPage.COUNT - 1)]

    companion object {
        fun preview(
            currentPage: Int = 0,
            hasCompletedOnboarding: Boolean? = false,
            isFinished: Boolean = false,
        ) = OnboardingUiState(
            currentPage = currentPage,
            hasCompletedOnboarding = hasCompletedOnboarding,
            isFinished = isFinished,
        )
    }
}

File: iosApp/iosApp/Features/Onboarding/OnboardingUiState.swift
import Foundation

struct OnboardingUiState: Equatable {
    var currentPage: Int = 0
    var hasCompletedOnboarding: Bool?
    var isFinished = false

    var shouldShowOnboarding: Bool {
        hasCompletedOnboarding == false && !isFinished
    }

    var currentPageData: OnboardingPage {
        let clamped = min(max(currentPage, 0), OnboardingPage.count - 1)
        return OnboardingPage(rawValue: clamped) ?? .linkTasks
    }

    static func preview(
        currentPage: Int = 0,
        hasCompletedOnboarding: Bool? = false,
        isFinished: Bool = false,
    ) -> OnboardingUiState {
        OnboardingUiState(
            currentPage: currentPage,
            hasCompletedOnboarding: hasCompletedOnboarding,
            isFinished: isFinished,
        )
    }
}
